contracts/OgvStaking.sol

```diff
@@ -172,24 +212,7 @@ contract OgvStaking is ERC20Votes {
     /// @param lockupId the id of the old lockup to extend
     /// @param duration number of seconds from now to stake for
     function extend(uint256 lockupId, uint256 duration) external {
-        // duration checked inside previewPoints
-        _collectRewards(msg.sender);
-        Lockup memory lockup = lockups[msg.sender][lockupId];
-        uint256 oldAmount = lockup.amount;
-        uint256 oldEnd = lockup.end;
-        uint256 oldPoints = lockup.points;
-        (uint256 newPoints, uint256 newEnd) = previewPoints(oldAmount, duration);
-        require(newEnd > oldEnd, "Staking: New lockup must be longer");
-        lockup.end = uint128(newEnd);
-        lockup.points = newPoints;
-        lockups[msg.sender][lockupId] = lockup;
-        _mint(msg.sender, newPoints - oldPoints);
-        if (!hasDelegationSet[msg.sender] && delegates(msg.sender) == address(0)) {
-            // Delegate voting power to the receiver, if unregistered
-            _delegate(msg.sender, msg.sender);
-        }
-        emit Unstake(msg.sender, lockupId, oldAmount, oldEnd, oldPoints);
-        emit Stake(msg.sender, lockupId, oldAmount, newEnd, newPoints);
+        revert StakingDisabled();
     }
 
     /// @notice Preview the number of points that would be returned for the
```

Comment:
previewPoints should revert with stakingDisabled.

Reply:
Done